File: src/test/java/com/wesleyhome/test/jupiter/MethodArgumentsProviderTest.java
package com.wesleyhome.test.jupiter;

import java.util.List;
import org.junit.jupiter.params.ParameterizedTest;

class MethodArgumentsProviderTest {

    @ParameterizedTest
    @ParametersSource
    void testParameters(Boolean bool) {
        System.out.println(bool);
    }

    @ParameterizedTest
    @ParametersSource
    void testParameters_primitive(boolean bool) {
        System.out.println(bool);
    }

    @ParameterizedTest
    @ParametersSource
    void testParameters_enum(TestEnum testEnum) {
        System.out.println(testEnum);
    }

    @ParameterizedTest
    @ParametersSource
    void testParameters_lots(TestEnum testEnum, Boolean wrap, boolean bool) {
        System.out.printf("%s - %s - %s%n", testEnum, wrap, bool);
    }

    @ParameterizedTest
    @ParametersSource({IntegerProvider.class})
    void testParameters_provider(int val) {
        System.out.printf("%d%n", val);
    }

    @ParameterizedTest
    @ParametersSource({IntegerProvider.class})
    void testParameters_providers(int val, double val2) {
        System.out.printf("%d - %f%n", val, val2);
    }

    @SuppressWarnings("WeakerAccess")
    public List<Double> doubles() {
        return List.of(2.0, 3.2, 5.6);
    }
}

File: src/test/kotlin/com/wesleyhome/test/jupiter/kotlin/MethodArgumentsProviderTest.kt
package com.wesleyhome.test.jupiter.kotlin

import com.wesleyhome.test.jupiter.IntegerProvider
import com.wesleyhome.test.jupiter.ParametersSource
import org.junit.jupiter.params.ParameterizedTest

internal class MethodArgumentsProviderTest {
    @ParameterizedTest
    @ParametersSource
    fun testParameters(bool: Boolean?) {
        println(bool)
    }

    @ParameterizedTest
    @ParametersSource
    fun testParameters_primitive(bool: Boolean) {
        println(bool)
    }

    @ParameterizedTest
    @ParametersSource
    fun testParameters_enum(testEnum: TestKotlinEnum?) {
        println(testEnum)
    }

    @ParameterizedTest
    @ParametersSource
    fun testParameters_enum_not_nullable(testEnum: TestKotlinEnum) {
        println(testEnum)
    }

    @ParameterizedTest
    @ParametersSource
    fun testParameters_lots(testEnum: TestKotlinEnum?, wrap: Boolean?, bool: Boolean) {
        System.out.printf("%s - %s - %s%n", testEnum, wrap, bool)
    }

    @ParameterizedTest
    @ParametersSource
    fun testParameters_lots_not_nullable(testEnum: TestKotlinEnum, wrap: Boolean?, bool: Boolean) {
        System.out.printf("%s - %s - %s%n", testEnum, wrap, bool)
    }

    @ParameterizedTest
    @ParametersSource(value = [IntegerProvider::class])
    fun testParameters_provider(`val`: Int) {
        System.out.printf("%d%n", `val`)
    }

    @ParameterizedTest
    @ParametersSource(value = [IntegerProvider::class])
    fun testParameters_providers(`val`: Int, val2: Double) {
        System.out.printf("%d - %f%n", `val`, val2)
    }

    fun doubles(): List<Double> {
        return listOf(2.0, 3.2, 5.6)
    }
}
